### PR TITLE
Create env vars to set JITaaS options

### PR DESCRIPTION
**Pull request by chrisc66**
WIP: Create env vars to set JITaaS options

[skip ci]

Besides command line options, environment variables are more convenient to use sometimes.
Added env vars in the format of `TR_JITaaSXXX`:
```
TR_JITaaSServerMode = 1
TR_JITaaSClientMode = 1
TR_JITaaSPort = XXX
TR_JITaaSServerAddress = XXX
TR_JITaaSTimeout = XXX
TR_JITaaSsslKey = XXX
TR_JITaaSsslCert = XXX
TR_JITaaSsslRootCerts = XXX
```

A few lines are commented out since it leads to compile time errors, the code is functionally working but has memory leaks. I am looking into it right now.

Resolves: #6019 

Signed-off-by: Chris <zichun.chong@mail.utoronto.ca>

### CODE REVIEW

File: runtime/compiler/control/J9Options.cpp [hunk 1]
```diff
@@ -2041,55 +2118,74 @@ J9::Options::fePreProcess(void * base)
...
+         }
+      // If no option, check for env var TR_JITaaSServerMode, TR_JITaaSClientMode
+      if (!hasModeOption)
+         {
+         char *envVarServerMode = feGetEnv("TR_JITaaSServerMode");
+         char *envVarClientMode = feGetEnv("TR_JITaaSClientMode");
+         if (envVarServerMode != NULL && *(envVarServerMode) == '1' && envVarClientMode != NULL && *(envVarClientMode) == '1')
             {
+            compInfo->getPersistentInfo()->setJITaaSMode(NONJITaaS_MODE);
+            printf("Both env vars are set, run in non-JITaaSMode. TR_JITaaSServerMode = 1, TR_JITaaSClientMode = 1\n");
```
Comment (chrisc66) [inline runtime/compiler/control/J9Options.cpp]: Is there any functions that we are using to produce message to user, other than printf?

File: runtime/compiler/control/J9Options.cpp [hunk 2]
```diff
@@ -1146,6 +1170,55 @@ static bool JITaaSParseOptionsCommon(char **options, char delimiter, TR::Compila
    return true;
    }
 
+static void JITaaSParseEnvVarCommon(TR::CompilationInfo *compInfo, const struct TR_JTIaaSCommonOptions & optionsSet)
+   {
+   
+   if (!optionsSet.hasAddrOption)
+      {
+      char* address = feGetEnv("TR_JITaaSServerAddress");
+      // j9mem_free_memory(address);
```
Comment (chrisc66) [inline runtime/compiler/control/J9Options.cpp]: Adding this line, `j9mem_free_memory`, into the code generates a compile time error that I have no idea what it means. It is related to the macros and probably the private port library? The error only happens in this function, i.e. I can call it from one layer above (where server/client mode is parsed) or from `JITaaSParseOptionsCommon` without having this error.

```
/jre/lib/amd64/compressedrefs/oti/j9port_generated.h:697:55:
error: ‘privatePortLibrary’ was not declared in this scope
#define j9mem_free_memory(param1) OMRPORT_FROM_J9PORT(privatePortLibrary)->mem_free_memory(OMRPORT_FROM_J9PORT(privatePortLibrary),param1)

/jre/lib/amd64/compressedrefs/oti/j9port.h:59:44: 
note: in definition of macro ‘OMRPORT_FROM_J9PORT’
#define OMRPORT_FROM_J9PORT(_j9portLib) (&(_j9portLib)->omrPortLibrary)

/runtime/compiler/control/J9Options.cpp:1179:7: 
note: in expansion of macro ‘j9mem_free_memory’
j9mem_free_memory(address);
```

**Comment by chrisc66** [comment]
I left two comments at line 1179 and line 2134, which are the two places I am not sure about. Could you please take a look? Thanks

File: runtime/compiler/control/J9Options.cpp [hunk 3]
```diff
@@ -2270,7 +2366,6 @@ J9::Options::setupJITaaSOptions()
       self()->setOption(TR_DisableEDO); // JITaaS limitation, EDO counters are not relocatable yet
       self()->setOption(TR_DisableKnownObjectTable);
       self()->setOption(TR_DisableMethodIsCold); // shady heuristic; better to disable to reduce client/server traffic
```
Comment (mpirvu) [inline runtime/compiler/control/J9Options.cpp]: make sure you ndon't delete this line or the CPU will increase

File: runtime/compiler/control/J9Options.cpp [hunk 4]
```diff
@@ -274,6 +274,17 @@ enum TR_XlpCodeCacheOptions
    XLPCC_PARSING_ERROR
    };
 
+// Helper routines to parse -XX:JITaaSServer / -XX:JITaaSClient options and env vars
```
Comment (mpirvu) [inline runtime/compiler/control/J9Options.cpp]: The comment says "routines" but this is a struct.

File: runtime/compiler/control/J9Options.cpp (hunk 2 above)
Comment (mpirvu) [reply runtime/compiler/control/J9Options.cpp]: You don't need to free the memory from feGetEnv.

File: runtime/compiler/control/J9Options.cpp [hunk 5]
```diff
@@ -1146,6 +1170,55 @@ static bool JITaaSParseOptionsCommon(char **options, char delimiter, TR::Compila
    return true;
    }
 
+static void JITaaSParseEnvVarCommon(TR::CompilationInfo *compInfo, const struct TR_JTIaaSCommonOptions & optionsSet)
```
Comment (mpirvu) [inline runtime/compiler/control/J9Options.cpp]: I would like a comment here describing what the function does: checks for env vars related to JITaaS and sets options accordingly. The optionSet specifies which JITaaS options have been already specified using command line options

File: runtime/compiler/control/J9Options.cpp (hunk 1 above)
Comment (mpirvu) [reply runtime/compiler/control/J9Options.cpp]: Search through this file for NLS

**Comment by mpirvu** [comment]
I checked how open-liberty works:
- If the user sets `JVM_ARGS` env var, those options will be passed to the java command line that starts liberty.
- If the user sets `IBM_JAVA_OPTIONS` env var, this will be appended to whatever liberty script sets for IBM_JAVA_OPTIONS.

In both cases the user could pass JITaaS options to the java process inside the liberty conatiner, so these new environment variables are not really needed (at least for liberty).

**Comment by chrisc66** [comment]
Yes, I guess setting `IBM_JAVA_OPTIONS` makes more sense then. We should follow the original  environment variable and leave the new ones for now.

**Comment by mpirvu** [comment]
Agreed. I'll close this PR

**Pull request closed without merging**